### PR TITLE
consolidate object::commit.rs with commit.rs

### PR DESCRIPTION
**Pull request by keepsimple1**
This is to solve issue https://github.com/radicle-dev/radicle-git/issues/52 .

The main observations:

- [`commit::Commit`](https://github.com/radicle-dev/radicle-git/blob/2bb35ed99640d960e277befd84b5c54b5dfc893b/radicle-surf/src/commit.rs#L105) is a superset of [`object::commit::Header`](https://github.com/radicle-dev/radicle-git/blob/2bb35ed99640d960e277befd84b5c54b5dfc893b/radicle-surf/src/object/commit.rs#L47). 

- [`commit::Author`](https://github.com/radicle-dev/radicle-git/blob/2bb35ed99640d960e277befd84b5c54b5dfc893b/radicle-surf/src/commit.rs#L39) is a superset of [`object::commit::Person`](https://github.com/radicle-dev/radicle-git/blob/2bb35ed99640d960e277befd84b5c54b5dfc893b/radicle-surf/src/object/commit.rs#L198).

- [`object::commit::Commit`](https://github.com/radicle-dev/radicle-git/blob/2bb35ed99640d960e277befd84b5c54b5dfc893b/radicle-surf/src/object/commit.rs#L36) stores too much state (namely `diff` and `branches`) in itself. These states can be retrieved on-demand using `Repository` methods.

Serialization backward compatibility: 

The main changes are: 
1. `sha1` is renamed to `id`.
2. New fields are included in JSON output:  `message` and `parents`.

### CODE REVIEW

File: radicle-surf/t/src/source.rs [hunk 1]
```diff
@@ -165,3 +174,20 @@ fn tree_ordering() {
...
+    let branches = repo.revision_branches(init_commit, glob).unwrap();
+
+    // branch: refs/heads/dev
+    // branch: refs/heads/master
+    // branch: refs/remotes/banana/orange/pineapple
+    // branch: refs/remotes/banana/pineapple
+    // branch: refs/remotes/origin/HEAD
+    // branch: refs/remotes/origin/dev
+    // branch: refs/remotes/origin/master
+    assert_eq!(branches.len(), 7);
```
Comment (FintanH) [inline radicle-surf/t/src/source.rs:192]: Why not just test the names instead of the length?

**Review by FintanH — APPROVED** [review]
Looks good other than the test case :+1:

**Comment by keepsimple1** [comment]
@FintanH I've updated the test case. Please review. A new approval is needed before it can merge. Thanks!

**Comment by FintanH** [comment]
Nice :+1: Merged :ok_hand:

**Pull request merged**